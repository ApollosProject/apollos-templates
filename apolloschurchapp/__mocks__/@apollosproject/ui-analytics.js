import React from 'react';

module.exports = {
  track: () => '',
  AnalyticsConsumer: ({ children }) => children({ test: jest.fn() }),
  AnalyticsProvider: ({ children }) => children,
  CoreNavigationAnalytics: ({ children }) =>
    children({ onNavigationStateChange: jest.fn() }),
  TrackEventWhenLoaded: () => null,
  withTrackOnPress: (Component) => (props) => <Component {...props} />,
};
