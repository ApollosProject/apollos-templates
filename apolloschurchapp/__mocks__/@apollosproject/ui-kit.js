module.exports = {
  ...jest.requireActual('@apollosproject/ui-kit'),
  NavigationService: {
    navigate: jest.fn(),
    setTopLevelNavigator: jest.fn(),
  },
};
