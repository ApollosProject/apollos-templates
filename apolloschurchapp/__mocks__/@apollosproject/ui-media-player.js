module.exports = {
  MediaPlayerSpacer: ({ children }) => children,
  MediaPlayer: () => 'MediaPlayer',
  MediaPlayerProvider: ({ children }) => children,
  playVideoMutation: 'mutation { playVideo }',
  withTabBarMediaSpacer: () => ({ children }) => children,
};
