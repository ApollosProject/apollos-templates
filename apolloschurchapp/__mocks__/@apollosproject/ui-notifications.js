module.exports = { NotificationsProvider: ({ children }) => children };
