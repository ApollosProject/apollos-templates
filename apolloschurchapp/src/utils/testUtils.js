/* eslint-disable import/prefer-default-export */
import renderer from 'react-test-renderer';
import wait from 'waait';

export const renderWithApolloData = async (component) => {
  const tree = renderer.create(component);
  await wait(1);
  tree.update(component);
  return tree;
};
