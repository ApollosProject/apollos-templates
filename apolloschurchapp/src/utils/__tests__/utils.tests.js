import React from 'react';
import { Text } from 'react-native';

import { fetchMoreResolver } from '@apollosproject/ui-connected';

import { renderWithApolloData } from '../testUtils';

describe('the fetchMoreResolver function', () => {
  const args = {
    collectionName: 'userFeed',
    variables: { first: 10, after: null },
    fetchMore: jest.fn(),
    data: {
      userFeed: {
        edges: [
          {
            node: { id: 'foo' },
          },
          {
            node: { id: 'bar' },
          },
        ],
        pageInfo: { endCursor: 'abc123' },
      },
    },
  };
  test('no endCursor', () => {
    expect(
      fetchMoreResolver({
        ...args,
        data: { userFeed: { edges: [], pageInfo: {} } },
      })()
    ).toBeUndefined();
  });
  test('combines data', () => {
    fetchMoreResolver({ ...args })();
    const previousResult = args.data;
    const fetchMoreResult = {
      userFeed: {
        edges: [{ node: { id: 'baz' } }],
        pageInfo: { endCursor: 'def456' },
      },
    };
    expect(
      args.fetchMore.mock.calls[0][0].updateQuery(previousResult, {
        fetchMoreResult,
      })
    ).toEqual({
      userFeed: {
        edges: [
          {
            node: { id: 'foo' },
          },
          {
            node: { id: 'bar' },
          },
          {
            node: { id: 'baz' },
          },
        ],
        pageInfo: { endCursor: 'def456' },
      },
    });
  });
});

test('renderWithApolloData renders the component', async () => {
  const tree = await renderWithApolloData(<Text>hello, world!</Text>);
  expect(tree).toMatchSnapshot();
});
