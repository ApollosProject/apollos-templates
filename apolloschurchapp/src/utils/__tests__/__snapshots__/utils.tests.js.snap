// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderWithApolloData renders the component 1`] = `
<Text>
  hello, world!
</Text>
`;
